### PR TITLE
add how to build a dapp tutorial

### PR DESCRIPTION
**Pull request by eshaben**
### Description/Original PRs

Original PR: https://github.com/moonbeam-foundation/moonbeam-docs/pull/612/files

### CODE REVIEW

File: tutorials/eth-api/how-to-build-a-dapp.md [hunk 1]
```diff
@@ -0,0 +1,1124 @@
...
+
+### 创建一个Hardhat项目 {: #create-hardhat-project }
+
+您可以使用以下指令在Hardhat上发起项目：
+
+```bash
+npx hardhat init
+```
+
+创建JavaScript或TypeScript Hardhat项目时，系统会询问您是否要安装示例项目的依赖项，即为安装Hardhat和[Hardhat Toolbox插件](https://hardhat.org/hardhat-runner/plugins/nomicfoundation -hardhat-toolbox#hardhat-toolbox){target=_blank}。您不需要工具箱中包含的所有插件，因此您可以安装Hardhat、Ethers和Hardhat Ethers插件，这就是本教程所需的全部内容：
```
Comment (themacexpert) [inline tutorials/eth-api/how-to-build-a-dapp.md:86]:
```suggestion
创建JavaScript或TypeScript Hardhat项目时，系统会询问您是否要安装示例项目的依赖项，即为安装Hardhat和[Hardhat Toolbox插件](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-toolbox#hardhat-toolbox){target=_blank}。您不需要工具箱中包含的所有插件，因此您可以安装Hardhat、Ethers和Hardhat Ethers插件，这就是本教程所需的全部内容：
```

File: tutorials/eth-api/how-to-build-a-dapp.md [hunk 2]
```diff
@@ -0,0 +1,1124 @@
...
+此范例为使用`0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac`账户的余额（在Moonbase Alpha中为DEV）。让我们解析其中的元素：
+
+- `jsonrpc` — JSON-RPC APR版本，通常为“2.0”
+- `id` — 用于定义请求回应的常数值，通常可以保持为
+- `method` — 特定从/至区块链读/写数据的函数。您可以在[我们的文档网站查看许多不同的RPC函数](/builders/get-started/eth-compare/rpc-support){target=_blank}
+- `params` — 根据特定`method`的输入参数阵列
+
+事实上还有其他额外的元素会被加入至JSON-RPC请求中，但这四个最为常见。
+
+在当前，这些JSON-RPC请求非常有用，但是在编写代码时，一遍又一遍地创建JSON对象可能会很麻烦。这就是为什么存在有助于抽象和促进这些请求的使用的库。Moonbeam提供了[许多库的文档](/builders/build/eth-api/libraries){target=_blank}，我们将在本教程中使用[Ethers.js](/builders/build/eth -api/libraries/ethersjs){target=_blank}。您仅需要了解，每当我们通过Ethers.js包与区块链交互时，我们实际上是在使用 JSON-RPC！
```
Comment (themacexpert) [inline tutorials/eth-api/how-to-build-a-dapp.md:64]:
```suggestion
在当前，这些JSON-RPC请求非常有用，但是在编写代码时，一遍又一遍地创建JSON对象可能会很麻烦。这就是为什么存在有助于抽象和促进这些请求的使用的库。Moonbeam提供了[许多库的文档](/builders/build/eth-api/libraries){target=_blank}，我们将在本教程中使用[Ethers.js](/builders/build/eth-api/libraries/ethersjs){target=_blank}。您仅需要了解，每当我们通过Ethers.js包与区块链交互时，我们实际上是在使用 JSON-RPC！
```

File: tutorials/eth-api/how-to-build-a-dapp.md [hunk 3]
```diff
@@ -0,0 +1,1124 @@
...
+};
+```
+
+现在屏幕右上角应该有一个按钮将您的钱包连接到您的前端！接下来，让我们了解如何从智能合约中读取数据。
+
+### 从智能合约中读取数据 {: #reading-from-contracts }
+
+只要我们知道我们想要读取什么，读取合约非常容易。对于我们的应用，我们将读取可以铸造的最大Token数量以及已铸造的Token数量。通过这种方式，我们可以向用户显示仍然可以铸造多少Token，并希望引起一些FOMO的注意……
+
+如果您只是使用JSON-RPC，您可以使用`eth_call`来获取此数据，但这样做非常困难，因为您必须[对您的请求进行编码](https://docs.soliditylang.org/en/latest /abi-spec.html){target=_blank}采用称为ABI编码的非直接方法。幸运的是，Ethers.js允许我们轻松创建以人类可读的方式表示合约的对象，只要我们拥有合约的ABI。我们在Hardhat项目的`artifacts`目录中拥有`MintableERC20.sol`合约的ABI、`MintableERC20.json`！
```
Comment (themacexpert) [inline tutorials/eth-api/how-to-build-a-dapp.md:450]:
```suggestion
如果您只是使用JSON-RPC，您可以使用`eth_call`来获取此数据，但这样做非常困难，因为您必须[对您的请求进行编码](https://docs.soliditylang.org/en/latest/abi-spec.html){target=_blank}采用称为ABI编码的非直接方法。幸运的是，Ethers.js允许我们轻松创建以人类可读的方式表示合约的对象，只要我们拥有合约的ABI。我们在Hardhat项目的`artifacts`目录中拥有`MintableERC20.sol`合约的ABI、`MintableERC20.json`！
```

**Review by themacexpert — APPROVED** [review]
The Build a dApp page is missing a disclaimer, other than that LGTM 👍

**Comment by eshaben** [comment]
good catches! thanks!!

**Pull request merged**